src/index.js

```diff
@@ -0,0 +1,233 @@
+};
+
+// reset city name section
+const updateCityNameReset = () => {
+    // resets city name and temperature
+    const resetCityInfo = () => {
+        cityNameHeader.textContent = defaultCityNameInput;
+        currentTemperature.value = defaultTempValue
+        tempValue.textContent = displayTemperature();
+        tempValue.style.color = "red";
+        cityNameTextBox.value = "";
+        updateWeatherDisplay();
+        updateBackground();
+    };
+
+    resetCityName.addEventListener("click", resetCityInfo);
+};
+
+// Get Temp of search terms
+const tempSearch = async (cityName) => {
+    const locationResponse = await axios.get(`${baseURL}/location`, {
+        params: {q: cityName}
+    });
+    const {lat, lon} = locationResponse.data[0];
+
+    const weatherResponse = await axios.get(`${baseURL}/weather`, {
+        params: {lat, lon}
+    });
+    currentTemperature.value = weatherResponse.data.main.temp;
+    tempValue.textContent = displayTemperature();
+    updateWeatherDisplay();
+};
+
+// - Changing the temperature should change the page's background.
+const updateBackground = () => {
+    const body = document.body;
+    let temp = kelvinToF(currentTemperature.value);
+    if (temp > 80) {
+        currentTemperature.image = "hot.png"
+    } else if (temp > 65) {
+        currentTemperature.image = "good.png"
+    } else {
+        currentTemperature.image = "cold.png"
+    }
+    body.style.backgroundImage = `url(./assets/tempretures/${currentTemperature.image})`;
+    body.style.backgroundSize = "cover";
+};
+
+// sky display
+
+const updateSkyDisplay = (selectedSky) => {
+    const skyDisplay = document.getElementById("sky");
+    skyDisplay.innerHTML = skies[selectedSky] || skies.chooseAForecast;
+};
+
+skySelect.addEventListener("change", (event) => {
+    const selectedSky = event.target.value;
+    const sky = document.getElementById("sky");
+    sky.innerHTML = "";
+    sky.textContent = skies[selectedSky];
+    updateSkyDisplay(selectedSky);
+});
+
+
+// set temp control
+increaseTempControl.addEventListener("click", () => {
+    if(currentTemperature.unit === "F"){
+        currentTemperature.value +=  0.555;
+    } else {
+        currentTemperature.value += 1;
+    }
+    tempValue.textContent = displayTemperature();
+    updateWeatherDisplay();
+});
+decreaseTempControl.addEventListener("click", () => {
+    if(currentTemperature.unit === "F") {
+        currentTemperature.value -=  0.555;
+    } else {
+        currentTemperature.value -= 1;
+    }
+    tempValue.textContent = displayTemperature();
+    updateWeatherDisplay();
+});
+// Get current temp
+currentTempButton.addEventListener("click", async () => {
+    const cityName = cityNameHeader.innerHTML;
+    await tempSearch(cityName)
+    updateWeatherDisplay();
+});
+
+citySearchButton.addEventListener("click", async () => {
+    const cityName = cityNameTextBox.value;
+    syncCityName()
+    await tempSearch(cityName)
+    updateWeatherDisplay();
+});
```

Comment:
Is there a way to reduce this duplicated code?

Could you write a helper function that takes in an event object, check what the event object is and then execute some code accordingly? Then register the event handler on the different buttons.

```js 
const getCurrentTemp(event) => {
  let cityName = '';
  if (event.target.id === 'currentTempButton') {
       cityName = cityNameHeader.innerHTML;
  } else {
        cityName = cityNameTextBox.value;
        syncCityName();
  }
  await tempSearch(cityName)
  updateWeatherDisplay();
};